transaction-impl/src/main/java/com/example/auction/transaction/impl/TransactionEntity.java

```diff
@@ -83,12 +83,31 @@ private Behavior negotiatingDelivery(TransactionState state) {
                 state().updateDeliveryPrice(evt.getDeliveryPrice())
         );
 
+        builder.setCommandHandler(ApproveDeliveryDetails.class, (cmd, ctx) -> {
+            if(cmd.getUserId().equals(state().getTransaction().get().getCreator())) {
+                if(state().getTransaction().get().getDeliveryData().isPresent() &&
+                        state().getTransaction().get().getDeliveryPrice().isPresent()) {
```

Comment:
A couple of code style comments:

1. `state().getTransaction().get()` appears several times, and clutters the code, making it harder to understand what each line is doing. It might be a good idea to extract a local `transaction` variable at the top of the command handler.

2. If you want to take it a step further, you could add some extra convenience methods to `TransactionState` and/or `Transaction` that encapsulate it further, such as `Transaction TransactionState.getCreator() { return getTransaction().get().getCreator(); }`, `boolean Transaction.hasDeliveryData()`, etc.

3. Or even better, you could create an higher level method like `Transaction.ensureCanApproveDeliveryDetails(cmd.getUserId())` that handles all of the if logic and exception handling, and call that before `ctx.thenPersist`. You'll know that if `ensureCanApproveDeliveryDetails` completes successfully, it's OK to persist.